### PR TITLE
Task/pas 63 add react query

### PR DESCRIPTION
**Pull request by lukszymanski**
Changes:
* added react-query
* update state for order

### CODE REVIEW

File: libs/api/data-transfer/src/lib/models/dish.model.ts [hunk 1]
```diff
@@ -8,3 +8,9 @@ export interface DishModel {
   updatedAt: string;
   user: UserModel;
 }
+
+export interface DishPurgatoryModal {
```
Comment (pawelkondraciuk) [inline libs/api/data-transfer/src/lib/models/dish.model.ts:12]: Modal?

File: libs/pages/order/src/lib/order-store/order.facade.tsx [hunk 2]
```diff
@@ -1,7 +1,150 @@
...
+    {
+      onMutate: async ({ dishId }) => {
+        await queryClient.cancelQueries(orderDishesKey);
+
+        const prevDishes =
+          queryClient.getQueryData<DishModel[]>(orderDishesKey);
+        queryClient.setQueryData<DishModel[]>(orderDishesKey, (prev = []) => {
+          const index = prev?.findIndex((item) => item.id === dishId);
+
+          return [...prev.slice(0, index), ...prev.slice(index + 1)];
```
Comment (pawelkondraciuk) [inline libs/pages/order/src/lib/order-store/order.facade.tsx:100]: why not `filter`?

File: libs/api/data-transfer/src/lib/models/dish.model.ts (hunk 1 above)
Comment (lukszymanski) [reply libs/api/data-transfer/src/lib/models/dish.model.ts:12]: removing this

File: libs/pages/order/src/lib/order-store/order.facade.tsx (hunk 2 above)
Comment (lukszymanski) [reply libs/pages/order/src/lib/order-store/order.facade.tsx:100]: it's just to update the item in same place without changing it's index

Comment (lukszymanski) [reply libs/pages/order/src/lib/order-store/order.facade.tsx]: ah yup, fixed

File: libs/pages/order/src/lib/order-dish/order-dish.tsx [hunk 3]
```diff
@@ -44,149 +46,166 @@ export function OrderDish({
     setIsDeleting(false);
   }, [inProgress]);
 
+  useEffect(() => {
+    setTimeout(() => {
```
Comment (pawelkondraciuk) [inline libs/pages/order/src/lib/order-dish/order-dish.tsx:50]: Dont forger to clear that timer

Comment (lukszymanski) [reply libs/pages/order/src/lib/order-dish/order-dish.tsx]: fixed

**Review by pawelkondraciuk — APPROVED** [review]

**Pull request merged**